pkg/config/mesh/networks_watcher.go

```diff
@@ -159,7 +159,7 @@ func (w *internalNetworkWatcher) DeleteNetworksHandler(registration WatcherHandl
 		return
 	}
 
-	slices.FilterInPlace(w.handlers, func(handler *watchHandler) bool {
+	w.handlers = slices.FilterInPlace(w.handlers, func(handler *watchHandler) bool {
```

Comment:
LGTM, but I think the assertion can be moved out.

Reply:
Good catch

Reply:
Why do we need to set it like this? I thought its mutated in place?

Reply:
oh for `s = s[:n]`. tricky!